cluster_utils/client/server_communication.py

```diff
@@ -5,24 +5,31 @@
 import pickle
 import socket
 import traceback
-
-import pyuv
+from typing import Any
 
 from cluster_utils.communication_server import MessageTypes
 
 from . import submission_state
 
 
-def send_message(message_type, message):
-    loop = pyuv.Loop.default_loop()
-    udp = pyuv.UDP(loop)
-    udp.try_send(
+def send_message(message_type: MessageTypes, message: Any) -> None:
```

Comment:
Indeed, with TCP, it will either hang or raise an Exception if the connection cannot be established, both of which are not good.  With UDP on the other hand, the client just sends the message out and doesn't care what happens with it, so it just proceeds execution.

I tend to agree that communication failures should not crash the user code (I didn't really think that through beforehand).  However, if we anyway ignore failures, then I don't see much benefit in TCP.  Maybe the possibility to detect/log failures but not sure if that is of any actual help to the user.

I would suggest the following:
- For now, switch back to the UDP implementation to keep the code shorter
- In the longer term, consider file-based communication to improve reliability

What do you think?

Reply:
Yes, I agree. The main advantage of TCP would be logging of failures. While this could be informative to us in some situations, e.g. to get some insight into those rare failure cases, I also think it's not worth the added overhead and code complexity.

Reply:
In any case, could you still add some exception handling in this function? Even the UDP approach can probably throw errors in some situations.

Reply:
Done.  I also added a timeout, just in case (not sure if this is actually relevant for UDP but probably doesn't hurt either way).